openmc/universe.py

```diff
@@ -124,6 +125,110 @@ def name(self, name):
+                    return [self, cell] + cell.fill.find(p)
+        return []
+
+    def plot(self, center=(0., 0., 0.), width=(1., 1.), pixels=(200, 200),
+             basis='xy', color_by='cell'):
+        """Display a slice plot of the universe.
+
+        Parameters
+        ----------
+        center : Iterable of float
+            Coordinates at the center of the plot
+        width : Iterable of float
+            Width of the plot in each basis direction
+        pixels : Iterable of int
+            Number of pixels to use in each basis direction
+        basis : {'xy', 'xz', 'yz'}
+            The basis directions for the plot
+        color_by : {'cell', 'material'}
+            Indicate whether the plot should be colored by cell or by material
+
+        """
+        import matplotlib.pyplot as plt
+
+        if basis == 'xy':
+            x_min = center[0] - 0.5*width[0]
+            x_max = center[0] + 0.5*width[0]
+            y_min = center[1] - 0.5*width[1]
+            y_max = center[1] + 0.5*width[1]
+        elif basis == 'yz':
+            # The x-axis will correspond to physical y and the y-axis will correspond to physical z
+            x_min = center[1] - 0.5*width[0]
+            x_max = center[1] + 0.5*width[0]
+            y_min = center[2] - 0.5*width[1]
+            y_max = center[2] + 0.5*width[1]
+        elif basis == 'xz':
+            # The y-axis will correspond to physical z
+            x_min = center[0] - 0.5*width[0]
+            x_max = center[0] + 0.5*width[0]
+            y_min = center[2] - 0.5*width[1]
+            y_max = center[2] + 0.5*width[1]
+
+        # Determine locations to determine cells at
+        x_coords = np.linspace(x_min, x_max, pixels[0], endpoint=False) + \
+                   0.5*(x_max - x_min)/pixels[0]
+        y_coords = np.linspace(y_max, y_min, pixels[1], endpoint=False) - \
+                   0.5*(y_max - y_min)/pixels[1]
+
+        colors = {}
+        img = np.zeros(pixels + (4,))  # Use RGBA form
+        for i, x in enumerate(x_coords):
+            for j, y in enumerate(y_coords):
+                if basis == 'xy':
+                    path = self.find((x, y, center[2]))
+                elif basis == 'yz':
+                    path = self.find((center[0], x, y))
+                elif basis == 'xz':
+                    path = self.find((x, center[1], y))
+
+                if len(path) > 0:
+                    try:
+                        if color_by == 'cell':
+                            uid = path[-1].id
+                        elif color_by == 'material':
+                            if path[-1].fill_type == 'material':
+                                uid = path[-1].fill.id
+                            else:
+                                continue
+                    except AttributeError:
+                        continue
+                    if uid not in colors:
+                        colors[uid] = (random.random(), random.random(),
```

Comment:
How about seeding the random number generator so that the colors are reproducible? And perhaps include a `seed` parameter to this method so that a user can easily toggle between different coloring schemes to find a suitable one?`